geopandas/array.py

```diff
@@ -1233,6 +1240,25 @@ def z(self):
             message = "z attribute access only provided for Point geometries"
             raise ValueError(message)
 
+    @property
+    def m(self):
+        """Return the m coordinate of point geometries in a GeoSeries"""
+        if not SHAPELY_GE_21:
+            raise ImportError("'m' requires shapely>=2.1.")
+
+        if (self.geom_type[~self.isna()] == "Point").all():
+            empty = self.is_empty
+            if empty.any():
+                nonempty = ~empty
+                coords = np.full_like(nonempty, dtype=float, fill_value=np.nan)
+                coords[nonempty] = shapely.get_m(self._data[nonempty])
```

Comment:
Side note: this makes me wonder if this is not something that we should also fix on the shapely side, to make the vectorized `get_m` (and other `get_..`) work with empty points without raising an error

Reply:
To be fair, I did not test this, just copied implementation of Z support.